src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_UsingStatement.cs

```diff
@@ -307,39 +307,39 @@ private BoundStatement RewriteUsingStatementTryFinally(SyntaxNode syntax, BoundB
             }
 
             BoundExpression disposeCall;
-
-            // PROTOTYPE: would this be better to do during binding? Then we just emit whatever bound method we found?
-            // if we have an explicit method reference, use that
-            // next, look to see if this is an async (need to confirm order here)
-            // try and get a non-async dipose next
-            // finally, fail
-
-            if (!(methodOpt is null))
+            if (methodOpt is null)
             {
-                // PROTOTYPE: pattern methods can't yet be async
-                disposeCall = methodOpt.IsExtensionMethod
-                    ? BoundCall.Synthesized(syntax, receiverOpt: null, methodOpt, local)
-                    : BoundCall.Synthesized(syntax, disposedExpression, methodOpt);
+                if (awaitOpt is null)
```

Comment:
>if (awaitOpt is null) [](start = 16, length = 21)

Consider extracting a private method to compact code in this method somewhat. Also, that will clarify the intent (ie. to set `methodOpt`)

Reply:
Done.